### PR TITLE
Add tBTC SDK Page

### PR DESCRIPTION
**Pull request by evandrosaturnino**
## Description

Add the tBTC SDK landing page to the website with multiple sections - Updated navbar to accommodate a new category for the upcoming Threshold products, starting with the tBTC SDK (thUSD and TACo).

- [x]  [Add tbtc sdk landing page](https://github.com/threshold-network/website/pull/105/commits/b03599a508d2e136e1d3ba80d6e45a74735c5679)
- [x]  [Fix integration component margin](https://github.com/threshold-network/website/pull/105/commits/5a7ff4c95bd5cf7cf8c434e5ffc9575ba180d9ee)
- [x]  [Refactor nav bar menu content](https://github.com/threshold-network/website/pull/105/commits/d681d1ec941e01b2464f49b6c1c28c309ea8a8fd)

Figma's design prototype: https://www.figma.com/file/5eN3hPMkCVOpXb5YQw6CtB/tBTC-SDK?type=design&node-id=1264%3A185&mode=design&t=btHMRHkFwM2x8nJN-1

## Notice

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes issue https://github.com/threshold-network/website/issues/90

## Testing

Please outline all testing steps
1. Clone the repo - git clone https://github.com/threshold-network/website.git
2. Install dependencies - yarn install
3. Run the app - yarn start

## Screenshots (if applicable)

image

image

image

For mobile:

image

### CODE REVIEW

File: src/templates/build-page/tbtc-sdk/FeaturesSection.tsx
Comment (michalsmiarowski) [inline src/templates/build-page/tbtc-sdk/FeaturesSection.tsx:15]: Unnecessary exports. We use those interfaces only in this file.

```suggestion
interface FeaturesSectionProps {
  cards: FeatureCardProps[]
  icon: ImageProps
}

interface FeatureCardProps {
  title: string
  description: string
  isBigSize?: boolean
}
```

File: src/templates/build-page/tbtc-sdk/FeaturesSection.tsx
Comment (michalsmiarowski) [inline src/templates/build-page/tbtc-sdk/FeaturesSection.tsx:52]: Let's add a typing here. We could probably also change the name from `resource` to `card`

A good rule of thumb is to avoid `any` type whenever possible.

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (michalsmiarowski) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx:16]: Actually those type are kinda related to `Card` component so maybe we could put them there instead of creating `src/types/index.ts` file.

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (michalsmiarowski) [inline src/templates/build-page/tbtc-sdk/UseCases.tsx:51]: The hover functionality is little bit over-engineered here IMO. We could just use `_hover` property from chakra. See: https://v1.chakra-ui.com/docs/styled-system/features/style-props#pseudo

This should produce the exact same outcome with less complexity:

```
      <Box
        ref={cardRef}
        rounded="2xl"
        p="2px"
        _hover={{
          bgGradient:
            "linear-gradient(180deg, rgba(189,48,255,0) 0%, rgba(125,0,255,1) 100%)",
        }}
      >
```

Don't forget to also remove the `useState` above if you agree with this.

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (michalsmiarowski) [inline src/templates/build-page/tbtc-sdk/UseCases.tsx:90]: Instead of moving the button left with the margin property we could just use `justifyContent`:

```suggestion
                  justifyContent={"left"}
```

Right now the button link has justi content set to "center" so that's why it has that left indent.

File: src/components/HighlightWord.tsx
Comment (michalsmiarowski) [inline src/components/HighlightWord.tsx:2]: Let's remove `Box` import from here

File: .gitignore
Comment (michalsmiarowski) [inline .gitignore:7]: Is this change necessary? It works fine for me without the slash at the beginning.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Good job overall 👍 Left some comments to look at before the merge though.

File: src/templates/home-page/SectionTemplate.tsx
Comment (evandrosaturnino) [reply src/templates/home-page/SectionTemplate.tsx:88]: Possibly a cache issue, This has been addressed earlier today in the commit: https://github.com/threshold-network/website/pull/105/commits/03bf45ca0bbc91612b582157f4f3f0db7a524f6e
The justify-content class has been added to enable the possibility of making the row of buttons centered for centered sections, like the one below: 
image

Comment (evandrosaturnino) [reply src/templates/home-page/SectionTemplate.tsx:88]: Currently, for the existing ones, it is looking like this:
image

File: .gitignore
Comment (evandrosaturnino) [reply .gitignore:7]: I've made this adjustment to the .gitignore file because it was ignoring all files in any 'build' folder. This change was necessary after adding the 'build' folder to the content folder for the build page addition. By inserting a vertical bar, I ensured that only the 'build' folder in the root directory is ignored, not the ones in other locations.

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/FeaturesSection.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/FeaturesSection.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/FeaturesSection.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/FeaturesSection.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/02522d97d220ee5e52ee32525e9d80228eec79d7

File: src/components/HighlightWord.tsx
Comment (evandrosaturnino) [reply src/components/HighlightWord.tsx]: Addressed in [Remove unused box element import](https://github.com/threshold-network/website/pull/105/commits/2af0b4d60eaeb72f830b13d42c90d84dbb78cb4d)

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (michalsmiarowski) [reply src/templates/build-page/tbtc-sdk/index.tsx]: The conversation is ocntinues in https://github.com/threshold-network/website/pull/105#discussion_r1434253601

File: src/templates/home-page/SectionTemplate.tsx
Comment (michalsmiarowski) [reply src/templates/home-page/SectionTemplate.tsx:88]: Ah yes, looks like it working now. Thanks for the exmplanation!

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (michalsmiarowski) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx]: That's awesome! But let's also remove `onMouseEnter` and `onMouseLeave` from the Box component and let's also remove `useState` above:

```
const [isHovered, setIsHovered] = useState(false)
```

File: .gitignore
Comment (michalsmiarowski) [reply .gitignore:7]: I see, thanks for the explanation!

File: src/types/index.ts
Comment (michalsmiarowski) [inline src/types/index.ts:11]:
```suggestion
  rowReverse: boolean
```

File: static/admin/config.yml
Comment (michalsmiarowski) [inline static/admin/config.yml:2562]: Shouldn't we also add `preTitle` to the `Use Cases Card`? It's present in `tbtc-sdk.md`:

https://github.com/threshold-network/website/blob/2af0b4d60eaeb72f830b13d42c90d84dbb78cb4d/src/content/pages/build/tbtc-sdk.md?plain=1#L25-L33

File: static/admin/config.yml
Comment (michalsmiarowski) [inline static/admin/config.yml:2673]: Is Pre Title necessary here? It looks like it's not used in `FeaturesCard`

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Some more comments down below

File: src/templates/home-page/SectionTemplate.tsx
Comment (michalsmiarowski) [inline src/templates/home-page/SectionTemplate.tsx:1]: Btw I've just noticed that in the figma the top navbar is redesigned a little bit:

image

It looks like we've added `tBTC app` button and removed icon link to threshold network forum.

Do we plan to address it in thi PR? @evandrosaturnino

Comment (evandrosaturnino) [reply src/templates/home-page/SectionTemplate.tsx:1]: Very good point, at first I was aiming to get that addressed in another PR, but after thinking more about this, considering that the menu is already being partially reordered I addressed this as well.

Implemented in [Add cta button in nav bar menu](https://github.com/threshold-network/website/pull/105/commits/896955ed28ede2a34fee5b687d842b7278edefe2)

File: static/admin/config.yml
Comment (evandrosaturnino) [reply static/admin/config.yml]: Addressed in https://github.com/threshold-network/website/pull/105/commits/881ed9ce6d620f650337130866fced4f60cc8105

File: static/admin/config.yml
Comment (evandrosaturnino) [reply static/admin/config.yml:2562]: Addressed in https://github.com/threshold-network/website/pull/105/commits/881ed9ce6d620f650337130866fced4f60cc8105

File: src/types/index.ts
Comment (evandrosaturnino) [reply src/types/index.ts]: Addressed in https://github.com/threshold-network/website/pull/105/commits/f6273f71a2ebc922af66ed7c6d36c28694e5a4e9

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx:64]: Addressed in https://github.com/threshold-network/website/pull/105/commits/8775651c5965910da9748b510fbd722cdebd9ea4

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/f6273f71a2ebc922af66ed7c6d36c28694e5a4e9

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/commit/f6273f71a2ebc922af66ed7c6d36c28694e5a4e9

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/98ae297fc835b2d216660f23ba1a1733e876a2f7

File: src/templates/build-page/tbtc-sdk/FeaturesSection.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/FeaturesSection.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/f0df2bb9fa34e2e9b15eec0799bc6e61eacd5487

File: src/templates/build-page/tbtc-sdk/FeaturesSection.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/FeaturesSection.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/f0df2bb9fa34e2e9b15eec0799bc6e61eacd5487

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/c13f9b5132f67d1efddaf45f0f01cbcec173b7f7

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/f6273f71a2ebc922af66ed7c6d36c28694e5a4e9

File: src/templates/build-page/tbtc-sdk/UseCases.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/UseCases.tsx]: Good catch, addressed in https://github.com/threshold-network/website/pull/105/commits/c13f9b5132f67d1efddaf45f0f01cbcec173b7f7

File: src/components/Navbar/index.tsx
Comment (michalsmiarowski) [inline src/components/Navbar/index.tsx:47]: Non-blocking:

I think that the `NavBar` would require some refactoring/touch-ups, because in some resolutions it looks off. For example this is how it looks for `1173px` width:

image

I think that setting the maxW to `1140px` with `justifyContent="space-between"` is a mistake. Content should probably just be justified to "center" and the width should be maxed out. But I'm not sure as of now.

I think this would require some more touch upd in the long run so I would say to address this issue in a separate PR. The bug is also happening on production (1126 px width):
image

but adding the dashboard link in the navigation made it a little bit worse (see previous screen).

To not block this PR, I will mark it as non-blocking. Addressing that issue in this PR might extend the time to merge this one, as more code equals to more potential bugs that can be found 😅  So I would say we separate that into another PR. Let's just not forget about that issue..

You can test it out yourself and see it you can reproduce the issue.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left few more comments to look at

File: src/components/Navbar/index.tsx
Comment (evandrosaturnino) [reply src/components/Navbar/index.tsx:47]: Good catch! I've opened an issue to keep track of that, will address it in a separate PR as advised. 
https://github.com/threshold-network/website/issues/107

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/8b7e3bfe75f7cbb85c07f28ef285c01cdecffaf3

File: src/templates/build-page/tbtc-sdk/index.tsx
Comment (evandrosaturnino) [reply src/templates/build-page/tbtc-sdk/index.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/8b7e3bfe75f7cbb85c07f28ef285c01cdecffaf3

File: src/templates/home-page/SectionTemplate.tsx
Comment (evandrosaturnino) [reply src/templates/home-page/SectionTemplate.tsx]: Addressed in https://github.com/threshold-network/website/pull/105/commits/e69a16e5d83486a6c994e200493399c04e59dd18

File: src/components/Navbar/index.tsx
Comment (michalsmiarowski) [reply src/components/Navbar/index.tsx:47]: Great! I'll keep this conversation open just in case

**Review by michalsmiarowski — APPROVED** [review]
LGTM

**Pull request merged**
